recipes/openblas/all/conanfile.py

```diff
@@ -28,66 +33,70 @@ class OpenblasConan(ConanFile):
         "use_thread": True,
         "dynamic_arch": False,
     }
-    generators = "cmake"
     short_paths = True
 
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
-    @property
-    def _build_subfolder(self):
-        return "build_subfolder"
-
-    def export_sources(self):
-        self.copy("CMakeLists.txt")
-
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
+            self.options.rm_safe("fPIC")
 
     def validate(self):
-        if hasattr(self, "settings_build") and tools.cross_building(self, skip_x64_x86=True):
+        if hasattr(self, "settings_build") and cross_building(self, skip_x64_x86=True):
             raise ConanInvalidConfiguration("Cross-building not implemented")
 
-    def source(self):
-        tools.get(
+    def source(self): 
+        get(self,
             **self.conan_data["sources"][self.version],
             strip_root=True,
-            destination=self._source_subfolder
+            destination=self.source_folder
         )
 
+        if Version(self.version) <= "0.3.15":
+            replace_in_file(self, os.path.join(self.source_folder, "cmake", "utils.cmake"),
+                            "set(obj_defines ${defines_in})", "set(obj_defines ${defines_in})\r\n\r\n" +
+                            "list(FIND obj_defines \"RC\" def_idx)\r\n" + "if (${def_idx} GREATER -1) \r\n\t" +
+                            "list (REMOVE_ITEM obj_defines \"RC\")\r\n\t" + "list(APPEND obj_defines  \"RC=RC\")\r\n" +
+                            "endif ()\r\n" + "list(FIND obj_defines \"CR\" def_idx)\r\n" +
+                            "if (${def_idx} GREATER -1) \r\n\t" + "list (REMOVE_ITEM obj_defines \"CR\")\r\n\t" +
+                            "list(APPEND obj_defines  \"CR=CR\")\r\n" + "endif ()")
```

Comment:
If this is a patch, I think it's worth putting it as a Conan one. Please, have a look at our [Policy about patching](https://github.com/conan-io/conan-center-index/blob/56c1638e25453cc8e2ce4a676604900ebc7b62a2/docs/adding_packages/sources_and_patches.md#policy-about-patching)

Reply:
Let's move forward with that because there are more recipes depending on this migration. Let's add this patch in another iteration 😉